packages/trader/src/Modules/Trading/Components/Form/ContractType/contract-type-widget.jsx

```diff
@@ -148,74 +136,70 @@ class ContractTypeWidget extends React.PureComponent {
+                onBackButtonClick={onBackButtonClick}
+                onClose={handleVisibility}
+                onChangeInput={onChangeInput}
+                onCategoryClick={handleCategoryClick}
             >
-                <ContractType.Display
-                    is_open={is_dialog_open || is_info_dialog_open}
-                    list={this.props.list}
-                    name={name}
-                    onClick={this.onWidgetClick}
-                    value={value}
-                />
-                <ContractType.Dialog
-                    is_info_dialog_open={is_info_dialog_open}
-                    is_open={is_dialog_open}
-                    item={item || { value }}
-                    categories={this.list_with_category}
-                    list={list}
-                    selected={this.state.selected_category || this.list_with_category[0].label}
-                    onBackButtonClick={this.onBackButtonClick}
-                    onClose={this.handleVisibility}
-                    onChangeInput={this.onChangeInput}
-                    onCategoryClick={this.handleCategoryClick}
-                >
-                    {!is_info_dialog_open ? (
-                        <ContractType.List
-                            handleInfoClick={this.handleInfoClick}
-                            handleSelect={this.handleSelect}
-                            is_equal={is_equal}
-                            list={this.selected_category_contracts}
-                            name={name}
-                            value={value}
-                        />
-                    ) : (
-                        <ContractType.Info
-                            handleNavigationClick={this.handleNavigationClick}
-                            initial_index={this.selected_contract_index}
-                            handleSelect={this.handleSelect}
-                            item={item || { value }}
-                            list={this.list_with_category}
-                        />
-                    )}
-                </ContractType.Dialog>
-            </div>
-        );
-    }
-}
+                {!is_info_dialog_open ? (
+                    <ContractType.List
+                        handleInfoClick={handleInfoClick}
+                        handleSelect={handleSelect}
+                        is_equal={is_equal}
+                        list={selected_category_contracts()}
+                        name={name}
+                        value={value}
+                    />
+                ) : (
+                    <ContractType.Info
+                        handleNavigationClick={handleNavigationClick}
+                        initial_index={selected_contract_index()}
+                        handleSelect={handleSelect}
+                        item={item || { value }}
+                        list={list_with_category()}
+                    />
+                )}
+            </ContractType.Dialog>
```

Comment:
Let's reverse this condition for better readability!
```suggestion
                {is_info_dialog_open ? (
                    ...
                ) : (
                    ...
                )}
            </ContractType.Dialog>
```

Reply:
yeah, you're right:+1: